### ISSUE TITLE
Need a bit clearer instructions

### ISSUE THREAD

**Issue by brentjohnston** [ShivamShrirao/diffusers#7]
Edit: Everything I posted below was before Nerdy Rodents video on Youtube. I was doing it all completely wrong haha. I left a comment below with some tips though because even with his video it took me 2 days to figure out, 3080 10gb here, works great!

OLD POST:

_Thanks for doing this, I have some feedback and issues though. The whole part in the instructions about "running the command" gave me an "export is not a recognized command". It took me forever to realize I only need to run the part "accelerate launch" and after that and adding those additional 2 parameters you mention in the beggining for 10gb vram (because I'm an idiot). But wanted to mention the launch.sh file does not match the instructions either and has different parameters. I tried to edit and use that but also get error. (Im using Windows 11/anaconda also btw)

I also having issues with the fact that I'm pretty new to this and I don't know how to format export MODEL_NAME="(MODEL LOCATION)"
If it's located in C:\models\model.ckpt do I put MODEL_NAME="C:\models\model.ckpt" or simply put "model.ckpt" or "./model.ckpt"

Basically just need ELI5 instructions for people like me. I tried all combination earlier but I still got this error:

The following values were not passed to `accelerate launch` and had defaults used instead:
        `--num_cpu_threads_per_process` was set to `12` to improve out-of-box performance
To avoid this warning pass in values for each of the problematic parameters or run `accelerate config`.
usage: train_dreambooth.py [-h] --pretrained_model_name_or_path PRETRAINED_MODEL_NAME_OR_PATH [--tokenizer_name TOKENIZER_NAME] --instance_data_dir INSTANCE_DATA_DIR [--class_data_dir CLASS_DATA_DIR] [--instance_prompt INSTANCE_PROMPT] [--class_prompt CLASS_PROMPT] [--with_prior_preservation] [--prior_loss_weight PRIOR_LOSS_WEIGHT]
                           [--num_class_images NUM_CLASS_IMAGES] [--output_dir OUTPUT_DIR] [--seed SEED] [--resolution RESOLUTION] [--center_crop] [--train_batch_size TRAIN_BATCH_SIZE] [--sample_batch_size SAMPLE_BATCH_SIZE] [--num_train_epochs NUM_TRAIN_EPOCHS] [--max_train_steps MAX_TRAIN_STEPS]
                           [--gradient_accumulation_steps GRADIENT_ACCUMULATION_STEPS] [--gradient_checkpointing] [--learning_rate LEARNING_RATE] [--scale_lr] [--lr_scheduler LR_SCHEDULER] [--lr_warmup_steps LR_WARMUP_STEPS] [--use_8bit_adam] [--adam_beta1 ADAM_BETA1] [--adam_beta2 ADAM_BETA2] [--adam_weight_decay ADAM_WEIGHT_DECAY]
                           [--adam_epsilon ADAM_EPSILON] [--max_grad_norm MAX_GRAD_NORM] [--push_to_hub] [--use_auth_token] [--hub_token HUB_TOKEN] [--hub_model_id HUB_MODEL_ID] [--logging_dir LOGGING_DIR] [--log_interval LOG_INTERVAL] [--mixed_precision {no,fp16,bf16}] [--not_cache_latents] [--local_rank LOCAL_RANK]
train_dreambooth.py: error: unrecognized arguments: \ \ \
Traceback (most recent call last):
  File "C:\Users\New\anaconda3\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\New\anaconda3\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\New\anaconda3\Scripts\accelerate.exe\__main__.py", line 7, in <module>
  File "C:\Users\New\anaconda3\lib\site-packages\accelerate\commands\accelerate_cli.py", line 43, in main
    args.func(args)
  File "C:\Users\New\anaconda3\lib\site-packages\accelerate\commands\launch.py", line 837, in launch_command
    simple_launcher(args)
  File "C:\Users\New\anaconda3\lib\site-packages\accelerate\commands\launch.py", line 354, in simple_launcher
    raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)
subprocess.CalledProcessError: Command '['C:\\Users\\New\\anaconda3\\python.exe', 'train_dreambooth.py', '--pretrained_model_name_or_path=model.ckpt', '--instance_data_dir=', '--output_dir=/dog2', '--instance_prompt=a photo of sks dog', '--resolution=512', '\\', '--train_batch_size=1', '\\', '--gradient_accumulation_steps=1', '--learning_rate=5e-6', '\\', '--lr_scheduler=constant', '--lr_warmup_steps=0', '--max_train_steps=400', '--gradient_checkpointing', '--use_8bit_adam']' returned non-zero exit status 2.

EDIT: Accelerate config part also needs instructions for the various popup menus, no idea what deepspeed is or whatever, and a bunch of the other stuff. Do I press 0 in the first part about no distrubuted training if I have a single GPU for instance? Would love to see an option to simply download the zip file instead of git also. Thanks!_

**Comment by rSh4d0w** [comment]
You have to put the path were your checkpoint is.
if your model lays in a folder called "output" which resides in the same folder as train_dreambooth.py you would set it to MODEL_NAME="output".
accelerate config, depends on your setup.
How many CPUs and GPUs do you have?
Select that what fits you.
If you only got one GPU/CPU you can't do distributed training, and choose simply 0.
For deepspeed simply choose NO. Unless you know what you do.
For further help, read the documentation at https://huggingface.co/docs/accelerate/v0.12.0/en/package_reference/cli#accelerate-config

**Comment by brentjohnston** [comment]
Thanks, Last question, how would I format the other line settings 
export INSTANCE_DIR="dog"
export OUTPUT_DIR="dog2"

if my dataset is under C:\dreambooth\dog and the output C:\dreambooth\dog2

I tried bunch of different combinations but still getting errors about CPU changing to 12 core, and line errors returned non-zero exit status

**Comment by brentjohnston** [comment]
Hey guys, I followed Nerdy Rodents [video here](https://www.youtube.com/watch?v=w6PTviOCYQY) and it solved all the issues. His video after that covers converting the output to a ckpt file also. I am on a 3080 10gb and 12 core 5900x and everything works great. 

Make sure you use his pastebin link for the cuda install stuff and not the nvidia site links stuff because it's newer there and not compatible with bitsandbytes yet. It took me forever to realize what was happening. Now if I could only figure out how to use this all offline because when I disconnect internet it gives huggingface token https error during training. I tried disabling the token stuff in the .sh file but it doesn't allow training then. I really don't want to be forced to be connected when uploading pics of myself, call me paranoid.

One more tip was to make sure your training .sh file you upload is formatted for unix before sending it over to ubuntu, I had no idea this was a thing. In notepad++ you go to Edit, EOL Conversion, and choose Unix (LF) Before Windows was formatting the empty space with hidden characters and training wouldn't work. Hope this helps!

**Comment by ZeroCool22** [comment]
> Hey guys, I followed Nerdy Rodents [video here](https://www.youtube.com/watch?v=w6PTviOCYQY) and it solved all the issues. His video after that covers converting the output to a ckpt file also. I am on a 3080 10gb and 12 core 5900x and everything works great.
> 
> Make sure you use his pastebin link for the cuda install stuff and not the nvidia site links stuff because it's newer there and not compatible with bitsandbytes yet. It took me forever to realize what was happening. Now if I could only figure out how to use this all offline because when I disconnect internet it gives huggingface token https error during training. I tried disabling the token stuff in the .sh file but it doesn't allow training then. I really don't want to be forced to be connected when uploading pics of myself, call me paranoid.
> 
> One more tip was to make sure your training .sh file you upload is formatted for unix before sending it over to ubuntu, I had no idea this was a thing. In notepad++ you go to Edit, EOL Conversion, and choose Unix (LF) Before Windows was formatting the empty space with hidden characters and training wouldn't work. Hope this helps!

Can you help me i'm having some problems.
